src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -734,13 +735,13 @@ public void optimize(OptimizeRequest optimize) throws ElasticsearchException {
     }
 
     public org.apache.lucene.util.Version minimumCompatibleVersion() {
-        org.apache.lucene.util.Version luceneVersion = org.apache.lucene.util.Version.LUCENE_3_6;
+        org.apache.lucene.util.Version luceneVersion = null;
         for(Segment segment : engine().segments()) {
-            if (luceneVersion.onOrAfter(segment.getVersion())) {
+            if (luceneVersion == null || luceneVersion.onOrAfter(segment.getVersion())) {
                 luceneVersion = segment.getVersion();
             }
         }
-        return luceneVersion;
+        return luceneVersion == null ?  Version.indexCreated(indexSettings).luceneVersion : luceneVersion;
     }
```

Comment:
if there are no docs in the index it won't do anything here yes! in this case we should make this method more low level - I will fix that too